lib/puppet_x/puppetlabs/influxdb/influxdb.rb

```diff
@@ -37,17 +40,42 @@ def initialize
       # Make class instance variables available as instance variables to whichever object calls this method
       # For subclasses which call super, the instance variables will be part of their scope
       def init_attrs(resources)
-        # TODO: Only one uri per resource type
+        # TODO: this can probably be refactored into a proper cache of resources
         resources.each do |resource|
           @host ||= resource[:host] ? resource[:host] : PuppetlabsInfluxdb.host
           @port ||= resource[:port] ? resource[:port] : PuppetlabsInfluxdb.port
           @use_ssl ||= (!resource[:use_ssl].nil?) ? resource[:use_ssl] : PuppetlabsInfluxdb.use_ssl
+          @use_system_store ||= resource[:use_system_store] ? resource[:use_system_store] : PuppetlabsInfluxdb.use_system_store
+          @ca_bundle ||= resource[:ca_bundle] ? resource[:ca_bundle] : PuppetlabsInfluxdb.ca_bundle
           @token ||= resource[:token]
           @token_file ||= resource[:token_file] ? resource[:token_file] : PuppetlabsInfluxdb.token_file
         end
 
         protocol = @use_ssl ? 'https' : 'http'
         @influxdb_uri = "#{protocol}://#{@host}:#{@port}"
+
+        if @use_system_store
```

Comment:
Thanks for implementing this! This logic shouldn't be necessary if you use `include_system_store` to load the puppet agent CA bundle and then in the puppet.conf set `ssl_trust_store` to include the certs into the puppet agent bundle. That way you don't need to manually set up a context and force users into the standardized Puppet agent `ssl_trust_store` pattern. 

Ref: https://www.rubydoc.info/gems/puppet/Puppet/HTTP/Client and https://www.puppet.com/docs/puppet/8/configuration.html#ssl-trust-store

Reply:
Would that mean we don't need the `ca_bundle` param at all because it can be added via `ssl_trust_store` in `puppet.conf`?  Or could the Ruby code do something with `ssl_trust_store` in case someone forgets to set it there?